### PR TITLE
elm-test-rs: init at 2.0

### PR DESCRIPTION
**Pull request by jpagex**
###### Motivation for this change
I would like to add [elm-test-rs](https://github.com/mpizenberg/elm-test-rs) to Nixpkgs.

This is an alternative to `elm-test` written in Rust by @mpizenberg.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@turboMaCk I took inspiration on what you have done for `elm-json`. Maybe you can double-check?
@mpizenberg Let me know if it is OK for you to publish on Nixpkgs.

### CODE REVIEW

**Comment by mpizenberg** [comment]
Yep I'm fine with this. Maybe @BrianHicks will be interested in this too since he had added a [`default.nix`](https://github.com/mpizenberg/elm-test-rs/blob/master/default.nix) to the project some time ago.

**Comment by turboMaCk** [comment]
@GrahamcOfBorg  build elmPackages.elm-test-rs

File: pkgs/development/compilers/elm/packages/elm-test-rs.nix
Comment (turboMaCk) [inline pkgs/development/compilers/elm/packages/elm-test-rs.nix:11]: I'm not sure if this is necessary. Have you tried to compile without pkg-config? I think rust relies on pkg-config only when linking to c libs which I don't think is done in case of this binary.

Comment (jpagex) [reply pkgs/development/compilers/elm/packages/elm-test-rs.nix:11]: I did not. I can try without it.

**Comment by jpagex** [comment]
> Yep I'm fine with this. Maybe @BrianHicks will be interested in this too since he had added a [`default.nix`](https://github.com/mpizenberg/elm-test-rs/blob/master/default.nix) to the project some time ago.

Sure! I've seen that the `default.nix` is not up-to-date, by the way.

File: pkgs/development/compilers/elm/packages/elm-test-rs.nix
Comment (mpizenberg) [reply pkgs/development/compilers/elm/packages/elm-test-rs.nix:11]: I don't know about pkg-config but I can confirm that elm-test-rs is compiled by default without any C dependency. You could have some if using openssl instead of rustls for the `ureq` dependency (https client)

Comment (jpagex) [reply pkgs/development/compilers/elm/packages/elm-test-rs.nix:11]: Indeed, it works without `pkg-config`. I will force-push to remove it.

Comment (jpagex) [reply pkgs/development/compilers/elm/packages/elm-test-rs.nix]: I removed it.

**Comment by turboMaCk** [comment]
> Sure! I've seen that the default.nix is not up-to-date, by the way.

@mpizenberg I think you should probably remove that file from the project unless you're willing to [maintain it](https://github.com/mpizenberg/elm-test-rs/blob/master/default.nix#L6-L10).

**Comment by turboMaCk** [comment]
@jpagex can you please split this PR into two separate commits as:

- first commit adding you to the maintainer list
- second commit `elm-test-rs: init at 2.0`

this will be required for committers to accept the PR. Other than that I don't have further comments. The binary builds correctly on my system (nixos x86_64).

**Comment by jpagex** [comment]
> @jpagex can you please split this PR into two separate commits as:
> 
>     * first commit adding you to the maintainer list
> 
>     * second commit `elm-test-rs: init at 2.0`
> 
> 
> this will be required for committers to accept the PR. Other than that I don't have further comments. The binary builds correctly on my system (nixos x86_64).

Yes, I'll do that!

**Review by turboMaCk — APPROVED** [review]
LGTM

**Comment by turboMaCk** [comment]
@domenkozar this looks good to me. Feel free to review and/or merge this once you have a minute.

**Comment by turboMaCk** [comment]
it seems I've just been nominated for committer so I will be able to merge this PR. Before I do that I would like to test just few more things that just came to my mind. I'll try to do that and merge this ASAP.

File: pkgs/development/compilers/elm/packages/elm-test-rs.nix
Comment (turboMaCk) [inline pkgs/development/compilers/elm/packages/elm-test-rs.nix:11]: I believe even curl dependency in here is not required. We should keep scope as small as possible so unless there is a reason to keep it I would remove it.

**Review by turboMaCk — COMMENTED** [review]
I did few more tests locally and it all seems work great. I just think there might be one more unnecessary dependency.

File: pkgs/development/compilers/elm/packages/elm-test-rs.nix
Comment (jpagex) [reply pkgs/development/compilers/elm/packages/elm-test-rs.nix:11]: I had tried removing it, but it did not work. Maybe I have done it wrong? Were you successful removing it from the deps?

Comment (jpagex) [reply pkgs/development/compilers/elm/packages/elm-test-rs.nix:11]: It is the first time I contribute to Nixpkgs, and I do not know very well what it is the best way to build locally. Maybe you have good pointers for that?

Comment (turboMaCk) [reply pkgs/development/compilers/elm/packages/elm-test-rs.nix:11]: from my shell:

```
marek@nixos-mainframe λ ~/Projects/nixpkgs (nixpkgs-review/1●)   
❯❯❯ <pkgs/development/compilers/elm/packages/elm-test-rs.nix | grep curl

marek@nixos-mainframe λ ~/Projects/nixpkgs (nixpkgs-review/1●)   
❯❯❯ nix-build -A elmPackages.elm-test-rs
error: anonymous function at /home/marek/Projects/nixpkgs/pkgs/development/compilers/elm/packages/elm-test-rs.nix:1:1 called with unexpected argument 'curl', at /home/marek/Projects/nixpkgs/pkgs/development/compilers/elm/default.nix:115:19

marek@nixos-mainframe λ ~/Projects/nixpkgs (nixpkgs-review/1●)   
❯❯❯ nix-build -A elmPackages.elm-test-rs
/nix/store/x4zi2khnwsix4l49d4hdn3hxzjn9z51q-elm-test-rs-2.0

marek@nixos-mainframe λ ~/Projects/nixpkgs (nixpkgs-review/1●)   
❯❯❯ ./result/bin/elm-test-rs --version
elm-test-rs 2.0.0
```

this is the patch:

```diff
diff --git a/pkgs/development/compilers/elm/default.nix b/pkgs/development/compilers/elm/default.nix
index 8b840f8945f..a4b0d12fd8b 100644
--- a/pkgs/development/compilers/elm/default.nix
+++ b/pkgs/development/compilers/elm/default.nix
@@ -113,7 +113,7 @@ let
     };
 
     elm-test-rs = import ./packages/elm-test-rs.nix {
-      inherit curl lib rustPlatform fetchurl openssl stdenv Security darwin;
+      inherit lib rustPlatform fetchurl openssl stdenv Security darwin;
     } // {
       meta = with lib; {
         description = "Fast and portable executable to run your Elm tests";
diff --git a/pkgs/development/compilers/elm/packages/elm-test-rs.nix b/pkgs/development/compilers/elm/packages/elm-test-rs.nix
index de06a1f4acd..a482809ab50 100644
--- a/pkgs/development/compilers/elm/packages/elm-test-rs.nix
+++ b/pkgs/development/compilers/elm/packages/elm-test-rs.nix
@@ -1,4 +1,4 @@
-{ lib, curl, rustPlatform, fetchurl, openssl, stdenv, Security, darwin }:
+{ lib, rustPlatform, fetchurl, openssl, stdenv, Security, darwin }:
 rustPlatform.buildRustPackage rec {
   pname = "elm-test-rs";
   version = "2.0";
@@ -8,7 +8,7 @@ rustPlatform.buildRustPackage rec {
     sha256 = "sha256:1manr42w613r9vyji7pxx5gb08jcgkdxv29qqylrqlwxa8d5dcid";
   };
 
-  buildInputs = [ curl openssl ] ++ lib.optionals stdenv.isDarwin [ Security darwin.apple_sdk.frameworks.CoreServices ];
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ Security darwin.apple_sdk.frameworks.CoreServices ];
 
   cargoSha256 = "sha256:1dpdlzv96kpc25yf5jgsz9qldghyw35x382qpxhkadkn5dryzjvd";
   verifyCargoDeps = true;

```

Comment (jpagex) [reply pkgs/development/compilers/elm/packages/elm-test-rs.nix]: I just force-pushed the version without `curl`.

**Review by turboMaCk — APPROVED** [review]
looks great! Thanks for contribution @jpagex and welcome to nixpkgs maintainers.

**Comment by jpagex** [comment]
> looks great! Thanks for contribution @jpagex and welcome to nixpkgs maintainers.

Thanks for your help and for your careful review! 😀

**Pull request merged**
